server/handlers/default.go

```diff
@@ -35,6 +37,76 @@ func MainHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) *e
 	return nil
 }
 
+// AtomicUpdateHandler will accept multiple TUF files and ensure that the storage
+// backend is atomically updated with all the new records.
+func AtomicUpdateHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) *errors.HTTPError {
+	defer r.Body.Close()
+	s := ctx.Value("metaStore")
+	if s == nil {
+		return &errors.HTTPError{
+			HTTPStatus: http.StatusInternalServerError,
+			Code:       9999,
+			Err:        fmt.Errorf("Version store is nil"),
+		}
+	}
+	store, ok := s.(storage.MetaStore)
+	if !ok {
+		return &errors.HTTPError{
+			HTTPStatus: http.StatusInternalServerError,
+			Code:       9999,
+			Err:        fmt.Errorf("Version store not configured"),
+		}
+	}
+	vars := mux.Vars(r)
+	gun := vars["imageName"]
+	reader, err := r.MultipartReader()
+	if err != nil {
+		return &errors.HTTPError{
+			HTTPStatus: http.StatusBadRequest,
+			Code:       9999,
+			Err:        err,
+		}
+	}
+	var updates []storage.MetaUpdate
+	for {
+		part, err := reader.NextPart()
+		if err == io.EOF {
+			break
+		}
+		role := part.FileName()
+		if role == "" {
```

Comment:
Certainly, but not in an undefined way. This will fail very gracefully. I'm a subscriber to the idea "if you ask me to do something but there's nothing to do, I'm going to tell you it's done" (i.e. return a 200)